File: snake-strategy-bono/src/main/java/co/electric/snake/strategy/bonostrategy/newdirectionprocessors/ByEquivalentBestDirections.java
package co.electric.snake.strategy.bonostrategy.newdirectionprocessors;

import co.electric.snake.framework.model.Direction;

public class ByEquivalentBestDirections extends NewDirectionProcessor {

    public ByEquivalentBestDirections(DependencyProvider dependencyProvider) {
        super(dependencyProvider);
    }

    @Override
    public Direction getNewDirection() {
        Direction newDirection = null;

        if (equivalentBestDirections != null && !equivalentBestDirections.isEmpty()) {
            newDirection = processFinalDirection(equivalentBestDirections);

            printer.print("Random element from the equivalent best directions: " + newDirection);
        }

        return newDirection;
    }

}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/newdirectionprocessor/ByEquivalentBestDirections.kt
package co.electric.snake.strategy.bonostrategy.newdirectionprocessor

import co.electric.snake.framework.model.Direction
import co.electric.snake.strategy.bonostrategy.BlockingDirectionContainer
import co.electric.snake.strategy.bonostrategy.SimpleDirectionContainer
import org.slf4j.LoggerFactory
import java.util.*

class ByEquivalentBestDirections : NewDirectionProcessor {

    companion object {
        private val LOG = LoggerFactory.getLogger(ByEquivalentBestDirections::class.java)
    }

    override val order = 4

    override fun process(filteredDirections: SimpleDirectionContainer, equivalentBestDirections: SimpleDirectionContainer, blockingDirections: BlockingDirectionContainer): Optional<Direction> {
        return processFinalDirection(equivalentBestDirections, LOG)
    }

}
